lisa/sut_orchestrator/libvirt/platform.py

```diff
@@ -161,6 +161,10 @@ def _delete_environment(self, environment: Environment, log: Logger) -> None:
         self._delete_nodes(environment, log)
```

Comment:
FYI: I believe `_delete_environment` is not called if `_deploy_environment` fails (throws an exception). For example, `_deploy_nodes` calls `_delete_nodes` in an `except` block. (Sorry, I should have remembered this for your previous change.)

Reply:
`_delete_environment` is being called when `_deploy_environment` throws an exception:
https://github.com/microsoft/lisa/blob/e4fc638c02fc27ce51439712ac1568a33c9f7cdc/lisa/runners/lisa_runner.py#L269-L295